### PR TITLE
Improvements of sqlitecachedbhandler tests

### PR DESCRIPTION
**Pull request by devos50**
`sqlitecachedbhandler.py` is probably the largest file that can be found in Tribler. In this PR, I improved the tests for this file. Since `sqlitecachedbhandler.py` is concerned with separate database handlers, I created separate files to test each database handler on its own.

I also did some other changes. I noticed that we were using quite big databases for the unit tests (around 10MB). They were zipped into a `.tar.gz` file. However, 99% of this database was unused during the unit tests. The tests for the upgrader only needed the version field of the database. So I replaced these databases with way smaller databases with some dummy data.

The coverage of the `Tribler/Core` module is around 50% now when running all tests in `Tribler/Test/Core`.

### CODE REVIEW

**Comment by devos50** [comment]
retest this please

**Comment by devos50** [comment]
retest this please

**Comment by whirm** [comment]
Regarding the smaller database: once that file got committed to git, it will take the same space even if you delete the file. I'd rather keep it and use it for future tests (for instance the database upgrade code is basically untouched by any test). So if you don't see any noticeable change in the test execution time, I'd rather drop that change.

File: Tribler/Test/Core/test_sqlitecachedb_upgrade.py [hunk 1]
```diff
@@ -34,27 +43,24 @@ def tearDown(self):
...
 
         db_migrator = DBUpgrader(self.session, self.sqlitedb, torrent_store=MockTorrentStore())
         self.assertRaises(VersionNoLongerSupportedError, db_migrator.start_migrate)
 
-    def test_upgrade_from_17(self):
-        pass
-        # TODO(emilon): Implement that one and 18 22 23
-        # assert sqlitedb.version == LATEST_DB_VERSION, "Database didn't get upgraded to latest version (%s != %s)" % (
-        #     sqlitedb.version, LATEST_DB_VERSION)
+    def test_upgrade_17_to_latest(self):
```
Comment (whirm) [inline Tribler/Test/Core/test_sqlitecachedb_upgrade.py]: I see you implemented that, so never mind my previous comment about that.

File: Tribler/Test/Core/test_sqlitecachedbhandler_torrents.py [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+        comments = 'something not inside'
+        assert m_comment.find(comments) == -1
+
+        m_trackers = self.tdb.getTrackerListByInfohash(m_infohash)
+        assert len(m_trackers) == 8
+        assert 'http://tpb.tracker.thepiratebay.org/announce' in m_trackers, m_trackers
+
+        s_torrent = self.tdb.getTorrent(s_infohash)
+        m_torrent = self.tdb.getTorrent(m_infohash)
+        assert s_torrent['name'] == 'Tribler_4.1.7_src.zip', s_torrent['name']
```
Comment (whirm) [inline Tribler/Test/Core/test_sqlitecachedbhandler_torrents.py]: Now that we are at it, could you replace the plain asserts with nose's ones?

**Comment by whirm** [comment]
I don't know what you added into the DB, but I think having a "real" Tribler DB is a good Idea for the upgrade tests.

File: Tribler/Test/Core/test_sqlitecachedb_upgrade.py (hunk 1 above)
Comment (devos50) [inline Tribler/Test/Core/test_sqlitecachedb_upgrade.py]: @whirm I can place the file back. Do you think it's ok if I use the smaller database for the unit tests then? Maybe we can use the bigger databases for other unit tests in that case.

**Comment by devos50** [comment]
@whirm I can write a second that that uses the 'old' bigger database I removed.

File: Tribler/Test/Core/test_sqlitecachedbhandler_torrents.py (hunk 2 above)
Comment (devos50) [inline Tribler/Test/Core/test_sqlitecachedbhandler_torrents.py]: Yeah I will do that.

**Comment by whirm** [comment]
What would be the use of the smaller one then? Wouldn't be bigger one supersede it?

**Comment by devos50** [comment]
@whirm yeah it would have no use then. I will copy the rows I created myself in the smaller database into the bigger database and use that database instead.

**Comment by devos50** [comment]
@whirm I replaced all built-in asserts with the assertions of nose. Moreover, I placed the bigger database back again and used that one for the unit tests. I noticed that the tar extract script was extracting to the tests directory so I changed that. I found out that I could replace the whole script with only a few lines of code. Now I'm first copying the `tar.gz` file to the session directory and I'm extracting it there.

**Comment by devos50** [comment]
@whirm I moved the `import tarfile` to the top and skipped the copying of the archive to the session directory.

**Comment by devos50** [comment]
retest this please

**Comment by devos50** [comment]
retest this please

**Comment by whirm** [comment]
Nice, I'll merge if everything works then

**Comment by devos50** [comment]
retest this please

**Comment by whirm** [comment]
re this: https://jenkins.tribler.org/job/GH_Tribler_PR_tests_linux/342/artifact/output/test_runners_output/00010.err/*view*/

I guess we will have to create a PR with debug prints all over in `_test_port`, `check_random_port` and `get_random_port` and make it fail.

**Comment by whirm** [comment]
retest this please

**Comment by devos50** [comment]
retest this please

**Comment by devos50** [comment]
@whirm tests are finally passing...

**Pull request merged**